build/build/src/ide/web/google_font.rs

```diff
@@ -156,6 +156,30 @@ pub async fn download_google_font(
     Ok(result)
 }
 
+pub async fn download_google_font_with_css(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    family: &str,
+    css_family: &str,
+    css_basepath: &str,
+    output_path: impl AsRef<Path>,
+    css_output_path: impl AsRef<Path>,
+) -> Result<Vec<PathBuf>> {
+    let paths = download_google_font(cache, octocrab, family, output_path).await?;
+    let mut css = String::new();
+    use std::fmt::Write;
+    for path in &paths {
+        let file = path.file_name().and_then(|name| name.to_str()).unwrap_or("");
+        writeln!(&mut css, "@font-face {{")?;
+        writeln!(&mut css, "  font-family: '{css_family}';")?;
+        writeln!(&mut css, "  src: url('{css_basepath}/{file}');")?;
+        writeln!(&mut css, "}}")?;
+        writeln!(&mut css)?;
```

Comment:
It might be worth to extract the CSS file generation logic, it is duplicated over several places and seems logically separate from downloading font.

Reply:
i've tried to refactor things out, but there are a number of small differences between most of them which makes it a bit difficult to make it properly generic

Reply:
Well, I haven't really delved into details. If this is as deduplicated as it makes sense than fine.
(though perhaps some comments / docs on the function could be useful to better notice the difference / what is actually specific)